### PR TITLE
feat: add subscriber preference dal section

### PR DESCRIPTION
**Pull request by djabarovgeorge**

### CODE REVIEW

File: libs/dal/src/repositories/subscriber-preference/subscriber-preference.schema.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+      push: {
+        type: Schema.Types.Boolean,
+        default: true,
+      },
+    },
+  },
+  schemaOptions
+);
+
+export interface IUserDocument extends SubscriberPreferenceEntity, Document {
```
Comment (ainouzgali) [inline libs/dal/src/repositories/subscriber-preference/subscriber-preference.schema.ts:44]: IUserDocument is an interesting choice 😉😉

Comment (ainouzgali) [inline libs/dal/src/repositories/subscriber-preference/subscriber-preference.schema.ts:44]: IUserDocument is an interesting choice 😉😉

File: libs/dal/src/repositories/subscriber-preference/subscriber-preference.entity.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+export class SubscriberPreferenceEntity {
+  _subscriberId: string;
+  _templateId: string;
+  enabled: boolean;
+  channels: { email: boolean; sms: boolean; in_app: boolean; direct: boolean; push: boolean };
```
Comment (ainouzgali) [inline libs/dal/src/repositories/subscriber-preference/subscriber-preference.entity.ts:5]: What do you think of creating a different interface for this? And creating this interface also in package shared ?

**Review by ainouzgali — APPROVED** [review]

File: libs/dal/src/repositories/subscriber-preference/subscriber-preference.schema.ts (hunk 1 above)
Comment (djabarovgeorge) [reply libs/dal/src/repositories/subscriber-preference/subscriber-preference.schema.ts:44]: Oh will fix shortly, thanks for noticing!!

File: libs/dal/src/repositories/subscriber-preference/subscriber-preference.entity.ts (hunk 2 above)
Comment (djabarovgeorge) [reply libs/dal/src/repositories/subscriber-preference/subscriber-preference.entity.ts:5]: Love the idea I was not sure if keep it that way or do Record<ChannelTypeEnum, boolean>

**Pull request merged**